File: backend/src/main/java/com/dto/query/ItemInRoom.java
package com.dto.query;

import org.springframework.beans.factory.annotation.Value;

public interface ItemInRoom {

    @Value("#{target.item_id}")
    int getId();

    @Value("#{target.room_id}")
    int getRoomId();

    @Value("#{target.item_name}")
    String getName();

    @Value("#{target.item_count}")
    int getItemCount();
}

File: frontend/src/app/model/item-in-room.ts
export class ItemInRoom {
  id: number;
  roomId: number;
  name: string;
  itemCount: number;

  constructor(itemInRoom: ItemInRoom) {
    this.id = itemInRoom.id;
    this.roomId = itemInRoom.roomId;
    this.name = itemInRoom.name;
    this.itemCount = itemInRoom.itemCount;
  }

  toStringNameAndCount() : string {
    return this.name + '(' + this.itemCount + ')';
  }
}
